### PR TITLE
about-me: migrate to GDBus

### PR DESCRIPTION
**Pull request by yetist**
Migrate from dbus-glib to gdbus.

### CODE REVIEW

File: capplets/about-me/mate-about-me-fingerprint.c [hunk 1]
```diff
@@ -182,16 +218,18 @@ set_fingerprint_label (GtkWidget *enable, GtkWidget *disable)
 		gtk_widget_hide (enable);
 		gtk_widget_show (disable);
 		is_disable = TRUE;
+    g_strfreev (fingers);
```
Comment (rbuj) [inline capplets/about-me/mate-about-me-fingerprint.c:221]: Memory leak if g_strv_length (fingers) is 0.

Example:
```
$ cat multidimensional-array.c 
/* gcc -fsanitize=leak -fno-omit-frame-pointer -g -ggdb3 -O0 `pkg-config --libs --cflags glib-2.0` multidimensional-array.c -o multidimensional-array */
#include <stdio.h>
#include <string.h>
#include <glib.h>
#include <glib.h>
int main() {
  char **arr = (char**) calloc(1, sizeof (char*));
  arr[0] = NULL;

  printf ("arr length is %d!\n", g_strv_length (arr));

  /* g_strfreev (arr); */

  return 0;
}
[robert@localhost ~]$ ./multidimensional-array 
arr length is 0!

=================================================================
==1357==ERROR: LeakSanitizer: detected memory leaks

Direct leak of 8 byte(s) in 1 object(s) allocated from:
    #0 0x7f2953e98842  (/lib64/liblsan.so.0+0x10842)
    #1 0x40115c in main /home/robert/multidimensional-array.c:7
    #2 0x7f2953bba041 in __libc_start_main ../csu/libc-start.c:308

SUMMARY: LeakSanitizer: 8 byte(s) leaked in 1 allocation(s).
```

File: capplets/about-me/mate-about-me-fingerprint.c [hunk 2]
```diff
@@ -140,9 +163,10 @@ get_error_dialog (const char *title,
 void
 set_fingerprint_label (GtkWidget *enable, GtkWidget *disable)
 {
-	char **fingers;
-	DBusGProxy *device;
+	gchar **fingers;
```
Comment (rbuj) [inline capplets/about-me/mate-about-me-fingerprint.c:166]: shadow variable from fingerprint-strings.h

File: capplets/about-me/mate-about-me-fingerprint.c [hunk 3]
```diff
@@ -388,39 +493,69 @@ assistant_prepare (GtkAssistant *ass, GtkWidget *page, EnrollData *data)
...
-		GValue value = { 0, };
-
-		if (!dbus_g_proxy_call (data->device, "Claim", &error, G_TYPE_STRING, "", G_TYPE_INVALID, G_TYPE_INVALID)) {
-			GtkWidget *d;
-			char *msg;
-
+    GVariant *ret;
+    GtkWidget *d;
+    char *msg;
+    gchar *error_name;
```
Comment (rbuj) [inline capplets/about-me/mate-about-me-fingerprint.c:503]: Shadowed declaration. See cppcheck report.

**Comment by rbuj** [comment]
You can remove `capplets/about-me/fprintd-marshal.list` file:
```patch
diff --git a/capplets/about-me/Makefile.am b/capplets/about-me/Makefile.am
index e9a9ebf..0af42ff 100644
--- a/capplets/about-me/Makefile.am
+++ b/capplets/about-me/Makefile.am
@@ -4,7 +4,7 @@ cappletname = about-me
 ui_files = mate-about-me-dialog.ui mate-about-me-password.ui mate-about-me-fingerprint.ui
 Desktop_in_files = mate-about-me.desktop.in
 
-BUILT_SOURCES = marshal.c marshal.h mate-about-me-resources.h mate-about-me-resources.c
+BUILT_SOURCES = mate-about-me-resources.h mate-about-me-resources.c
 nodist_mate_about_me_SOURCES = \
        $(BUILT_SOURCES)
 mate_about_me_SOURCES =                \
@@ -17,10 +17,6 @@ mate_about_me_SOURCES =              \
        fingerprint-strings.h           \
        mate-about-me.c
 
-marshal.h: fprintd-marshal.list
-       @GLIB_GENMARSHAL@ --prefix=fprintd_marshal $< --header > $@
-marshal.c: fprintd-marshal.list
-       @GLIB_GENMARSHAL@ --prefix=fprintd_marshal $< --body --prototypes > $@
 mate-about-me-resources.h mate-about-me-resources.c: org.mate.mcc.am.gresource.xml Makefile $(shell $(GLIB_COMPILE_RESOURCES) --generate-dependencies --sourcedir $(srcdir) $(srcdir)/org.mate.mcc.am.gresource.xml)
        $(AM_V_GEN) XMLLINT=$(XMLLINT) $(GLIB_COMPILE_RESOURCES) --target $@ --sourcedir $(srcdir) --generate --c-name about_me $<
 
@@ -56,7 +52,6 @@ endif
 CLEANFILES = $(MATECC_CAPPLETS_CLEANFILES) $(desktop_DATA) $(BUILT_SOURCES)
 EXTRA_DIST = \
        $(ui_files) \
-       fprintd-marshal.list \
        $(Desktop_in_files) \
        org.mate.mcc.am.gresource.xml \
        ui-a11y.suppr
```

File: capplets/about-me/mate-about-me-fingerprint.c [hunk 4]
```diff
@@ -388,39 +493,69 @@ assistant_prepare (GtkAssistant *ass, GtkWidget *page, EnrollData *data)
...
+                                  "Get",
+                                  g_variant_new ("(ss)",
+                                                 "net.reactivated.Fprint.Device",
+                                                 "num-enroll-stages"),
+                                  G_DBUS_CALL_FLAGS_NONE,
+                                  -1,
+                                  NULL,
+                                  &error);
+    g_object_unref (p);
+    if ( ret == NULL) {
```
Comment (rbuj) [inline capplets/about-me/mate-about-me-fingerprint.c:557]: there is an extra space

File: capplets/about-me/mate-about-me-fingerprint.c [hunk 5]
```diff
@@ -534,16 +707,45 @@ enroll_fingerprints (GtkWindow *parent, GtkWidget *enable, GtkWidget *disable)
...
-			data->is_swipe = TRUE;
-		g_hash_table_destroy (props);
+  ret = g_dbus_proxy_call_sync (p,
+                                "GetAll",
+                                g_variant_new ("(s)", "net.reactivated.Fprint.Device"),
+                                G_DBUS_CALL_FLAGS_NONE,
+                                -1,
+                                NULL,
+                                NULL);
+  if ( ret != NULL) {
```
Comment (rbuj) [inline capplets/about-me/mate-about-me-fingerprint.c:732]: there is an extra space

**Comment by sc0w** [comment]
New warnings detected in the logs:

```
mate-about-me-fingerprint.c:94:7: warning: variable 'device_str' is used uninitialized whenever 'if' condition is true [-Wsometimes-uninitialized]
  if (ret == NULL) {
      ^~~~~~~~~~~
mate-about-me-fingerprint.c:114:11: note: uninitialized use occurs here
  g_free (device_str);
          ^~~~~~~~~~
mate-about-me-fingerprint.c:94:3: note: remove the 'if' if its condition is always false
  if (ret == NULL) {
  ^~~~~~~~~~~~~~~~~~
mate-about-me-fingerprint.c:84:19: note: initialize the variable 'device_str' to silence this warning
  char *device_str;
                  ^
                   = NULL

mate-about-me-fingerprint.c:102:12: warning: 5th function call argument is an uninitialized value
  device = g_dbus_proxy_new_for_bus_sync (G_BUS_TYPE_SYSTEM,
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

mate-about-me-fingerprint.c:715:5: warning: Potential leak of memory pointed to by 'data'
    g_warning ("Unable to contact Fprint Device daemon: %s\n", error->message);
    ^~~~~~~~~

mate-about-me-fingerprint.c:735:69: warning: Casting a non-structure type to a structure type and accessing a field can lead to memory access errors or data corruption
      if (g_str_equal (key, "name") && g_variant_is_of_type (value, G_VARIANT_TYPE_STRING)) {
                                                                    ^~~~~~~~~~~~~~~~~~~~~
/usr/include/glib-2.0/glib/gvarianttype.h:122:46: note: expanded from macro 'G_VARIANT_TYPE_STRING'
#define G_VARIANT_TYPE_STRING               ((const GVariantType *) "s")
                                             ^~~~~~~~~~~~~~~~~~~~~~~~~~

mate-about-me-fingerprint.c:737:81: warning: Casting a non-structure type to a structure type and accessing a field can lead to memory access errors or data corruption
      } else if (g_str_equal (key, "scan-type") && g_variant_is_of_type (value, G_VARIANT_TYPE_STRING)) {
                                                                                ^~~~~~~~~~~~~~~~~~~~~
/usr/include/glib-2.0/glib/gvarianttype.h:122:46: note: expanded from macro 'G_VARIANT_TYPE_STRING'
#define G_VARIANT_TYPE_STRING               ((const GVariantType *) "s")
                                             ^~~~~~~~~~~~~~~~~~~~~~~~~~

mate-about-me-fingerprint.c:411:24: warning: unused parameter 'proxy' [-Wunused-parameter]
on_signal (GDBusProxy *proxy,
                       ^

mate-about-me-fingerprint.c:412:24: warning: unused parameter 'sender_name' [-Wunused-parameter]
           gchar      *sender_name,
                       ^
```

File: capplets/about-me/mate-about-me-fingerprint.c [hunk 6]
```diff
@@ -500,13 +671,13 @@ assistant_prepare (GtkAssistant *ass, GtkWidget *page, EnrollData *data)
...
 enroll_fingerprints (GtkWindow *parent, GtkWidget *enable, GtkWidget *disable)
 {
-	DBusGProxy *device, *p;
-	GHashTable *props;
+	GDBusProxy *device;
 	GtkBuilder *dialog;
 	EnrollData *data;
 	GtkWidget *ass;
 	char *msg;
 	GError *error = NULL;
```
Comment (raveit65) [inline capplets/about-me/mate-about-me-fingerprint.c:679]: @yetist 
Indent..

File: capplets/about-me/mate-about-me-fingerprint.c [hunk 7]
```diff
@@ -456,42 +600,69 @@ assistant_prepare (GtkAssistant *ass, GtkWidget *page, EnrollData *data)
...
+                                  G_DBUS_CALL_FLAGS_NONE,
+                                  -1,
+                                  NULL,
+                                  &error);
+    if (ret == NULL) {
 			/* translators:
 			 * The variable is the name of the device, for example:
 			 * "Could you not access "Digital Persona U.are.U 4000/4000B" device */
 			msg = g_strdup_printf (_("Could not start finger capture on '%s' device"), data->name);
-			d = get_error_dialog (msg, dbus_g_error_get_name (error), GTK_WINDOW (data->ass));
```
Comment (raveit65) [inline capplets/about-me/mate-about-me-fingerprint.c:477]: Indent...

File: capplets/about-me/mate-about-me-fingerprint.c [hunk 8]
```diff
@@ -456,42 +600,69 @@ assistant_prepare (GtkAssistant *ass, GtkWidget *page, EnrollData *data)
...
+      error_name = g_dbus_error_get_remote_error (error);
+			d = get_error_dialog (msg, error_name, GTK_WINDOW (data->ass));
 			g_error_free (error);
 			gtk_dialog_run (GTK_DIALOG (d));
 			gtk_widget_destroy (d);
 			g_free (msg);
```
Comment (raveit65) [inline capplets/about-me/mate-about-me-fingerprint.c:631]: Indent...

File: capplets/about-me/mate-about-me-fingerprint.c [hunk 9]
```diff
@@ -456,42 +600,69 @@ assistant_prepare (GtkAssistant *ass, GtkWidget *page, EnrollData *data)
...
+      g_free (error_name);
 
 			enroll_data_destroy (data);
 
 			return;
 		}
 		data->state = STATE_ENROLLING;;
```
Comment (raveit65) [inline capplets/about-me/mate-about-me-fingerprint.c:638]: For me the whole block until line 666 has Indents errors.

File: capplets/about-me/mate-about-me-fingerprint.c [hunk 10]
```diff
@@ -140,9 +166,10 @@ get_error_dialog (const char *title,
 void
 set_fingerprint_label (GtkWidget *enable, GtkWidget *disable)
 {
-	char **fingers;
-	DBusGProxy *device;
+  gchar **enrolled_fingers;
```
Comment (raveit65) [inline capplets/about-me/mate-about-me-fingerprint.c:169]: Indents..

File: capplets/about-me/mate-about-me-fingerprint.c [hunk 11]
```diff
@@ -183,15 +222,16 @@ set_fingerprint_label (GtkWidget *enable, GtkWidget *disable)
 		gtk_widget_show (disable);
 		is_disable = TRUE;
 	}
-
-	g_strfreev (fingers);
+  g_strfreev (enrolled_fingers);
 	g_object_unref (device);
```
Comment (raveit65) [inline capplets/about-me/mate-about-me-fingerprint.c:226]: Indent

File: capplets/about-me/mate-about-me-fingerprint.c [hunk 12]
```diff
@@ -245,12 +297,31 @@ delete_fingerprints_question (GtkBuilder *dialog, GtkWidget *enable, GtkWidget *
 static void
 enroll_data_destroy (EnrollData *data)
 {
+  GVariant *ret;
```
Comment (raveit65) [inline capplets/about-me/mate-about-me-fingerprint.c:300]: Indents

File: capplets/about-me/mate-about-me-fingerprint.c [hunk 13]
```diff
@@ -245,12 +297,31 @@ delete_fingerprints_question (GtkBuilder *dialog, GtkWidget *enable, GtkWidget *
...
+                                  g_variant_new ("()"),
+                                  G_DBUS_CALL_FLAGS_NONE,
+                                  -1,
+                                  NULL,
+                                  NULL);
+    if (ret != NULL) {
+      g_variant_unref (ret);
+    }
 		/* fall-through */
 	case STATE_CLAIMED:
```
Comment (raveit65) [inline capplets/about-me/mate-about-me-fingerprint.c:314]: Indent

File: capplets/about-me/mate-about-me-fingerprint.c [hunk 14]
```diff
@@ -245,12 +297,31 @@ delete_fingerprints_question (GtkBuilder *dialog, GtkWidget *enable, GtkWidget *
...
+                                  g_variant_new ("()"),
+                                  G_DBUS_CALL_FLAGS_NONE,
+                                  -1,
+                                  NULL,
+                                  NULL);
+    if (ret != NULL) {
+      g_variant_unref (ret);
+    }
 		/* fall-through */
 	case STATE_NONE:
```
Comment (raveit65) [inline capplets/about-me/mate-about-me-fingerprint.c:326]: Indent

File: capplets/about-me/mate-about-me-fingerprint.c [hunk 15]
```diff
@@ -340,10 +411,23 @@ assistant_cancelled (GtkAssistant *ass, EnrollData *data)
...
 static void
-enroll_result (GObject *object, const char *result, gboolean done, EnrollData *data)
+on_signal (GDBusProxy *proxy G_GNUC_UNUSED,
+           gchar      *sender_name G_GNUC_UNUSED,
+           gchar      *signal_name,
+           GVariant   *parameters,
+           EnrollData *data)
 {
-	GtkBuilder *dialog = data->dialog;
 	char *msg;
```
Comment (raveit65) [inline capplets/about-me/mate-about-me-fingerprint.c:420]: Indent

File: capplets/about-me/mate-about-me-fingerprint.c [hunk 16]
```diff
@@ -340,10 +411,23 @@ assistant_cancelled (GtkAssistant *ass, EnrollData *data)
...
+
+  if (!g_str_equal (signal_name, "EnrollStatus")) {
+    return;
+  }
+
+  GtkBuilder *dialog = data->dialog;
+
+  g_variant_get (parameters, "(sb)", &result, &done);
 
 	if (g_str_equal (result, "enroll-completed") || g_str_equal (result, "enroll-stage-passed")) {
```
Comment (raveit65) [inline capplets/about-me/mate-about-me-fingerprint.c:432]: Indents.
All 3 `if` blocks starting at the wrong place for me.

File: capplets/about-me/mate-about-me-fingerprint.c [hunk 17]
```diff
@@ -355,27 +439,50 @@ enroll_result (GObject *object, const char *result, gboolean done, EnrollData *d
...
+                                  "EnrollStop",
+                                  g_variant_new ("()"),
+                                  G_DBUS_CALL_FLAGS_NONE,
+                                  -1,
+                                  NULL,
+                                  NULL);
+    if (ret != NULL) {
+      g_variant_unref (ret);
+    }
 		data->state = STATE_CLAIMED;
```
Comment (raveit65) [inline capplets/about-me/mate-about-me-fingerprint.c:460]: Same issue here.

File: capplets/about-me/mate-about-me-fingerprint.c [hunk 18]
```diff
@@ -355,27 +439,50 @@ enroll_result (GObject *object, const char *result, gboolean done, EnrollData *d
...
+      result = g_strdup ("enroll-retry-scan");
 		} else {
+      g_free (result);
 			return;
 		}
 	}
 
 	msg = g_strdup_printf (TR(enroll_result_str_to_msg (result, data->is_swipe)), data->name);
 	gtk_label_set_text (GTK_LABEL (WID ("status-label")), msg);
 	g_free (msg);
```
Comment (raveit65) [inline capplets/about-me/mate-about-me-fingerprint.c:484]: Indent

File: capplets/about-me/mate-about-me-fingerprint.c [hunk 19]
```diff
@@ -388,39 +495,69 @@ assistant_prepare (GtkAssistant *ass, GtkWidget *page, EnrollData *data)
...
 		GError *error = NULL;
 		GtkBuilder *dialog = data->dialog;
 		char *path;
 		guint i;
-		GValue value = { 0, };
-
-		if (!dbus_g_proxy_call (data->device, "Claim", &error, G_TYPE_STRING, "", G_TYPE_INVALID, G_TYPE_INVALID)) {
-			GtkWidget *d;
-			char *msg;
-
```
Comment (raveit65) [inline capplets/about-me/mate-about-me-fingerprint.c:401]: Line 491 to 501 has other indents....

File: capplets/about-me/mate-about-me-fingerprint.c [hunk 20]
```diff
@@ -388,39 +495,69 @@ assistant_prepare (GtkAssistant *ass, GtkWidget *page, EnrollData *data)
...
+                                  -1,
+                                  NULL,
+                                  &error);
+    if (error != NULL) {
 			/* translators:
 			 * The variable is the name of the device, for example:
 			 * "Could you not access "Digital Persona U.are.U 4000/4000B" device */
 			msg = g_strdup_printf (_("Could not access '%s' device"), data->name);
-			d = get_error_dialog (msg, dbus_g_error_get_name (error), GTK_WINDOW (data->ass));
+      error_name = g_dbus_error_get_remote_error (error);
```
Comment (raveit65) [inline capplets/about-me/mate-about-me-fingerprint.c:520]: Indents

File: capplets/about-me/mate-about-me-fingerprint.c [hunk 21]
```diff
@@ -388,39 +495,69 @@ assistant_prepare (GtkAssistant *ass, GtkWidget *page, EnrollData *data)
...
+			d = get_error_dialog (msg, error_name, GTK_WINDOW (data->ass));
 			g_error_free (error);
 			gtk_dialog_run (GTK_DIALOG (d));
 			gtk_widget_destroy (d);
 			g_free (msg);
+      g_free (error_name);
```
Comment (raveit65) [inline capplets/about-me/mate-about-me-fingerprint.c:526]: Indents

File: capplets/about-me/mate-about-me-fingerprint.c [hunk 22]
```diff
@@ -388,39 +495,69 @@ assistant_prepare (GtkAssistant *ass, GtkWidget *page, EnrollData *data)
...
 
 			enroll_data_destroy (data);
 
 			return;
-		}
+		} else {
+      g_variant_unref (ret);
```
Comment (raveit65) [inline capplets/about-me/mate-about-me-fingerprint.c:532]: Indents

File: capplets/about-me/mate-about-me-fingerprint.c [hunk 23]
```diff
@@ -432,20 +569,27 @@ assistant_prepare (GtkAssistant *ass, GtkWidget *page, EnrollData *data)
 
 			enroll_data_destroy (data);
 
-			g_object_unref (p);
+      g_error_free (error);
```
Comment (raveit65) [inline capplets/about-me/mate-about-me-fingerprint.c:572]: Line 561 to 571 has othe indents.

**Comment by yetist** [comment]
First, I use the follow commands to check the code style under capplets, and I found more files use 2 space indent:

```
$ find capplets/ -name "*.c" -exec cat -T {} \;|grep "^ "
```
Then, I follow it like this:

https://github.com/mate-desktop/mate-control-center/blob/ecf39a01667e3f54b4128990832e35cbcdeae013/capplets/about-me/mate-about-me-fingerprint.c#L2

So, I use 2 spaces for the new code line, but the origenal file use Tab. If you use vim to open the file, you can get a better alignment effect. 

I was told that the new code should use space indent and not Tab. The original `mate-about-me-fingerprint.c` file uses Tab indent, but I use 2 spaces, if use other text editor to open it, code indent is wrong.

I really want to suggest to do a code indent for the whole project, just like this commit did, https://github.com/mate-desktop/mozo/commit/80873ff2ec320d0a4feffdec97e86d626e2cf7db

**Comment by raveit65** [comment]
Well, you've got your reasons to do it like you did. But does this looks correct and nice?

>  if use other text editor to open it, code indent is wrong.

I reviewed everything in browser on github site, here it looks wrong.

**Comment by raveit65** [comment]
Only one example:
```
@@ -140,9 +166,10 @@ get_error_dialog (const char *title,
 void
 set_fingerprint_label (GtkWidget *enable, GtkWidget *disable)
 {
-	char **fingers;
-	DBusGProxy *device;
+  gchar **enrolled_fingers;
+  GDBusProxy *device;
 	GError *error = NULL;
+  GVariant *ret;
 
 	gtk_widget_set_no_show_all (enable, TRUE);
 	gtk_widget_set_no_show_all (disable, TRUE);
```
Does this looks good?

Why not using 4 spaces at the beginning of a line?

> Then, I follow it like this:

https://github.com/mate-desktop/mate-control-center/blob/ecf39a01667e3f54b4128990832e35cbcdeae013/capplets/about-me/mate-about-me-fingerprint.c#L2

This was written by a gnome developer 20 years ago :)

**Comment by yetist** [comment]
> Only one example:
> Does this looks good?

No, so the new code should be use Tab to keep the style, or use 8 spaces, not 4 space or 2 space.

> 
> Why not using 4 spaces at the beginning of a line?
> This was written by a gnome developer 20 years ago :)

In general there are two preferred indentation styles for code in GNOME.
1. Linux Kernel style. Tabs with a length of 8 characters are used for the indentation, with K&R brace placement.
2. GNU style. Each new level is indented by 2 spaces, braces go on a line by themselves, and they are indented as well.

No 4 spaces on GNOME.

**Comment by yetist** [comment]
Will using 4 spaces make a difference? It looks as bad as 2 spaces.
```
@@ -140,9 +166,10 @@ get_error_dialog (const char *title,
 void
 set_fingerprint_label (GtkWidget *enable, GtkWidget *disable)
 {
-       char **fingers;
-       DBusGProxy *device;
+    gchar **enrolled_fingers;
+    GDBusProxy *device;
        GError *error = NULL;
+    GVariant *ret;
 
        gtk_widget_set_no_show_all (enable, TRUE);
        gtk_widget_set_no_show_all (disable, TRUE);
```

File: capplets/about-me/mate-about-me-fingerprint.c [hunk 24]
```diff
@@ -166,10 +166,10 @@ get_error_dialog (const char *title,
 void
 set_fingerprint_label (GtkWidget *enable, GtkWidget *disable)
 {
-  gchar **enrolled_fingers;
-  GDBusProxy *device;
+    gchar **enrolled_fingers;
+    GDBusProxy *device;
 	GError *error = NULL;
```
Comment (raveit65) [inline capplets/about-me/mate-about-me-fingerprint.c:171]: What is so hard to change line 171 that it looks like the other lines?

**Comment by raveit65** [comment]
@yetist 
1 We are not using tabs, see our wiki tabs.
2. My world is so beautiful without discussion about badly code style, so i am stepping back from the review.
3. Maybe another team member like to merge PRs with bad code-style?
4. Better review another PRs from other people, only opening PRs is not enough.

**Comment by raveit65** [comment]
If it's easier for you than fix the formatting-style of the whole file with with a second commit here in PR.

**Comment by yetist** [comment]
I really don't know what to do?
1. Respect the old style and still use Tab indentation.
2. Use the new style and indent with 4 spaces, which is what this PR does.
3. Submit 2 times in the PR, the first time only convert the code style to the new format, and the second time modify the code.

**Comment by sc0w** [comment]
@yetist 

> Respect the old style and still use Tab indentation

yes, it is easy to do

if you like to change the style with 4 spaces it is not required here and it can be done in new PR in the whole file, but it is another story unrelated to `"migrate to GDBus"`

File: capplets/about-me/mate-about-me-fingerprint.c [hunk 25]
```diff
@@ -56,40 +54,67 @@ typedef struct {
...
+						"net.reactivated.Fprint.Device",
+						NULL,
+						&error);
+	if (device == NULL) {
+		g_warning ("Unable to contact Fprint Manager daemon: %s\n", error->message);
+		g_error_free (error);
+		goto out;
+	}
 
+out:
```
Comment (sc0w) [inline capplets/about-me/mate-about-me-fingerprint.c:116]: `out` is needed ? it seems it goes always to next line without `goto`

**Comment by sc0w** [comment]
new warnings in the logs:

```
mate-about-me-fingerprint.c:735:67: warning: Casting a non-structure type to a structure type and accessing a field can lead to memory access errors or data corruption
                        if (g_str_equal (key, "name") && g_variant_get_type (value) == G_VARIANT_TYPE_STRING) {
                                                                                       ^~~~~~~~~~~~~~~~~~~~~

mate-about-me-fingerprint.c:737:79: warning: Casting a non-structure type to a structure type and accessing a field can lead to memory access errors or data corruption
                        } else if (g_str_equal (key, "scan-type") && g_variant_get_type (value) == G_VARIANT_TYPE_STRING) {
                                                                                                   ^~~~~~~~~~~~~~~~~~~~~
```

File: capplets/about-me/mate-about-me-fingerprint.c [hunk 26]
```diff
@@ -499,64 +497,64 @@ assistant_prepare (GtkAssistant *ass, GtkWidget *page, EnrollData *data)
...
+					   &error);
+		if (p == NULL) {
+			g_warning ("Unable to contact Fprint Device daemon: %s\n", error->message);
+			g_error_free (error);
+			return;
+		}
+		ret = g_dbus_proxy_call_sync (p,
+					      "Get",
+					      g_variant_new ("(ss)",
+						      "net.reactivated.Fprint.Device",
```
Comment (raveit65) [inline capplets/about-me/mate-about-me-fingerprint.c:548]: Indent.

**Comment by raveit65** [comment]
@yetist 
Can you please squash all commits in one?
Then i can start test the final commit.

**Comment by raveit65** [comment]
Any open questions or can this be merged?

**Comment by sc0w** [comment]
https://github.com/mate-desktop/mate-control-center/pull/579#issuecomment-674113668

https://travis-ci.org/github/mate-desktop/mate-control-center/builds/718204086

```
mate-about-me-fingerprint.c:735:67: warning: Casting a non-structure type to a structure type and accessing a field can lead to memory access errors or data corruption
                        if (g_str_equal (key, "name") && g_variant_get_type (value) == G_VARIANT_TYPE_STRING) {
                                                                                       ^~~~~~~~~~~~~~~~~~~~~

mate-about-me-fingerprint.c:737:79: warning: Casting a non-structure type to a structure type and accessing a field can lead to memory access errors or data corruption
                        } else if (g_str_equal (key, "scan-type") && g_variant_get_type (value) == G_VARIANT_TYPE_STRING) {
                                                                                                   ^~~~~~~~~~~~~~~~~~~~~
```

**Comment by sc0w** [comment]
@yetist see https://github.com/mate-desktop/mate-control-center/pull/585

**Pull request merged**